### PR TITLE
highlight the need for `weave expose` in CNI/k8s

### LINKED ISSUE #2405: Unable to create a pod on kubernetes cluster using weave
**Setup details:**
Three vagrant machines.
1) kubernetes master --> running etcd, apiserver, scheduler, controller-manager
2) kubernetes host h1 --> weave launch
3) kubernetes host h2 --> weave launch <IP address of h1>

**OS/Environment:**
- Docker 1.10.3.
- Ubuntu trusty 14.04
- Weave 1.6

**Command run:**
`./kubectl create -f simple_nginx_pod.yml`

**Expected result:**
The pods get created successfully.

**Error:**
```
E0627 21:34:43.691382   24255 cni.go:167] Error adding network: Bridge "weave" has no IP addresses
E0627 21:34:43.691601   24255 cni.go:118] Error while adding to cni network: Bridge "weave" has no IP addresses
E0627 21:34:43.691932   24255 manager.go:1874] Failed to setup network for pod "nginx_default(f6366379-3cae-11e6-818d-08002707be12)" using network plugins "cni": Bridge "weave" has no IP addresses; Skipping pod
E0627 21:34:43.703957   24255 pod_workers.go:138] Error syncing pod f6366379-3cae-11e6-818d-08002707be12, skipping: failed to "SetupNetwork" for "nginx_default" with SetupNetworkError: "Failed to setup network for pod \"nginx_default(f6366379-3cae-11e6-818d-08002707be12)\" using network plugins \"cni\": Bridge \"weave\" has no IP addresses; Skipping pod"
```

I followed the exact steps mentioned in the docs to start weave.

```
vagrant@ubuntuh1:~$ sudo docker ps -a
CONTAINER ID        IMAGE                        COMMAND                  CREATED             STATUS              PORTS               NAMES
44e0739cbdeb        weaveworks/plugin:1.6.0      "/home/weave/plugin"     7 minutes ago       Up 1 seconds                            weaveplugin
2e2f1749e906        weaveworks/weaveexec:1.6.0   "/home/weave/weavepro"   7 minutes ago       Up 1 seconds                            weave proxy
02c8b683d9e6        weaveworks/weaveexec:1.6.0   "/bin/false"             7 minutes ago       Created                                 weavevolumes-1.6.0
84604b7f9839        weaveworks/weave:1.6.0       "/home/weave/weaver -"   7 minutes ago       Up 1 seconds                            weave
16cf231f83bb        weaveworks/weavedb           "data-only"              7 minutes ago       Created                                 waved
vagrant@ubuntuh1:~$
vagrant@ubuntuh1:~$ sudo weave status

        Version: 1.6.0 (up to date; next check at 2016/06/28 02:44:16)

        Service: router
       Protocol: weave 1..2
           Name: ae:10:77:d2:aa:77(ubuntuh1)
     Encryption: disabled
  PeerDiscovery: enabled
        Targets: 1
    Connections: 1 (1 established)
          Peers: 2 (with 2 established connections)
 TrustedSubnets: none

        Service: spam
         Status: ready
          Range: 10.32.0.0/12
  DefaultSubnet: 10.32.0.0/12

        Service: dns
         Domain: weave.local.
       Upstream: 10.0.2.3
            TTL: 1
        Entries: 0

        Service: proxy
        Address: unix:///var/run/weave/weave.sock

        Service: plugin
     DriverName: weave

vagrant@ubuntuh1:~$
```

### PR DESCRIPTION
**Pull request by rade**
Fixes #2405

### CODE REVIEW

**Comment by rade** [comment]
cc @abuehrle

**Comment by bboreham** [comment]
I might want to stress that the container network is opened up to the host network; that is really what K8s needs, and the user might not realise that is what `weave expose` does.

**Comment by rade** [comment]
I was taking my cue from the k8s error message `Bridge "weave" has no IP addresses`

**Comment by bboreham** [comment]
That error message is from our CNI plugin: https://github.com/weaveworks/weave/blob/master/plugin/net/cni.go#L142

It is trying to set up the route in and out of the container.

**Comment by rade** [comment]
ah ok. Tweaked.

Should we change the error message perhaps? To say something like "... Did you forget to run 'weave expose'?"

**Comment by bboreham** [comment]
Error message changed

File: site/cni-plugin.md [hunk 1]
```diff
@@ -30,12 +30,16 @@ To create a network which can span multiple hosts, the Weave peers must be conne
...
 See [Using Weave Net](/site/using-weave.md#peer-connections) for a discussion on peer connections. 
 
-As well as launching Weave Net, you have to run an extra command to
-perform some additional configuration of the Weave bridge:
-
     weave launch <peer hosts>
+
+Then you have to run an extra command to perform some additional
+configuration required by Kubernetes:
+
```
Comment (abuehrle) [inline site/cni-plugin.md]: What extra command do you need to run?  I find this a little confusing.  Do you need to run the extra command before you run `weave expose` and if so what does the extra command do and why do I need it?

Comment (rade) [inline site/cni-plugin.md]: `weave expose` *is* the extra command.

Comment (abuehrle) [inline site/cni-plugin.md]: But what is the `additional configuration required by Kubernetes`?  That's what's not clear.

Comment (rade) [inline site/cni-plugin.md]: "assigning an IP address to the Weave bridge and enabling access to containers from the host".

Am tweaking it...

Comment (abuehrle) [inline site/cni-plugin.md]: Ok, I'll edit that file a bit tomorrow.

Comment (rade) [inline site/cni-plugin.md]: tweaked

**Pull request merged**